### PR TITLE
Issues #ED-94 and #ED-618 feat: Block level filter in program dashboard and Accessing program level reports

### PR DESCRIPTION
**Pull request by 5Amogh**
# SunbirdEd - Portal
---

### Block level filter in program dashboard and Accessing program level reports
---

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts [hunk 1]
```diff
@@ -33,4 +31,15 @@ export class SbTableComponent implements AfterViewInit  {
     this.lib.instance.reset();
     this.loadTable();
   }
-}
+
+  getTableData() {
+    return [{ id: this.config.id , data: this.rowsData , selectedFilters: this.currentFilters }];
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts:36]: what is config is empty please use the 
? perator

File: src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts [hunk 2]
```diff
@@ -33,4 +31,15 @@ export class SbTableComponent implements AfterViewInit  {
...
+  }
+
+  filterChanged(data: any): void {
+    this.currentFilters = data.filters;
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts:40]: same as above

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts [hunk 3]
```diff
@@ -104,7 +112,31 @@ export class DatasetsComponent implements OnInit, OnDestroy {
   displayFilters:any = {};
   loadash = _;
   pdFilters:ConfigFilter[] = [];
-  configuredFilters:any = {}
+  configuredFilters:any = {};
+  appliedFilters:object = {};
+  blocks:object[] = [
+    {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:118]: why is there a data hardcoded
and why in sunbird this code is there?

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts [hunk 4]
```diff
@@ -260,26 +302,44 @@ export class DatasetsComponent implements OnInit, OnDestroy {
         return data;
       }
     });
+    console.log('program',program)
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:305]: why do we have a console .log

**Review by rajeevsathish — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts (hunk 2 above)
Comment (5Amogh) [reply src/app/client/src/app/modules/dashboard/components/sb-table/sb-table.component.ts:40]: Sure these changes will be made, thanks

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts (hunk 3 above)
Comment (5Amogh) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:118]: Apologies, had put in for testing, will remove this, thanks.

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.spec.ts [hunk 5]
```diff
@@ -771,12 +847,146 @@ describe('DatasetsComponent', () => {
...
+    component.selectedReport = mockData.selectedReportForSolutionTest
+    jest.spyOn(component,'checkStatus');
+    component.checkStatus();
+    expect(component.checkStatus).toHaveBeenCalled();
+  });
+
+  it('should call checkStatus with program', () => {
+    component.selectedSolution = undefined;
+    jest.spyOn(onDemandReportService,'isInProgress').mockReturnValue(true)
+    component.onDemandReportData = [{
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.spec.ts:873]: can't you put this json in data object and get it from there?

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts [hunk 6]
```diff
@@ -734,7 +803,7 @@ export class DatasetsComponent implements OnInit, OnDestroy {
       this.filter = [];
       setTimeout(() => {
         this.isProcessed = false;
-      }, 5000);
+      }, 10000);
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:806]: why do you need a 10 sec delay?

**Review by rajeevsathish — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts (hunk 6 above)
Comment (5Amogh) [reply src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts:806]: earlier the user was seeing a particular message which depends on this flag for just 5 secs although message required some 6-7 seconds normally to read so had to update this to 10 seconds

**Pull request merged**